src/Umbraco.Core/Services/DataTypeService.cs

```diff
@@ -434,6 +434,37 @@ private void ConvertMissingEditorsOfDataTypesToLabels(IEnumerable<IDataType> dat
             return OperationResult.Attempt.Succeed(MoveOperationStatusType.Success, evtMsgs);
         }
 
+        public Attempt<OperationResult<MoveOperationStatusType, IDataType>?> Copy(IDataType copying, int containerId)
+        {
+            var evtMsgs = EventMessagesFactory.Get();
+
+            IDataType copy;
+            using (var scope = ScopeProvider.CreateCoreScope())
+            {
+                try
+                {
+                    if (containerId > 0)
+                    {
+                        var container = _dataTypeContainerRepository.Get(containerId);
+                        if (container == null)
+                            throw new DataOperationException<MoveOperationStatusType>(MoveOperationStatusType.FailedParentNotFound); // causes rollback
```

Comment:
Nitpicking here but we should use pattern matching and braces style 👮 
```suggestion
                        if (container is not null)
                        {
                            throw new DataOperationException<MoveOperationStatusType(MoveOperationStatusType.FailedParentNotFound); // causes rollback
                        }
```

Reply:
Updated!